tests/virt_autotest/virsh_external_snapshot.pm

```diff
@@ -47,14 +46,15 @@ sub run_test {
     #Wait for forceful shutdown of active guests
     sleep 60;
 
-    my $get_vm_hostnames_inactive   = "virsh list --inactive | grep -E \"${vm_types}\" | awk \'{print \$2}\'";
-    my $vm_hostnames_inactive       = script_output($get_vm_hostnames_inactive, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my $vm_hostnames_inactive       = script_output("virsh list --inactive --name", $wait_script, type_command => 0, proceed_on_failure => 0);
     my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
 
     foreach my $guest (keys %virt_autotest::common::guests) {
         my $type = check_guest_disk_type($guest);
         next if ($type == 1);
         record_info "virsh-snapshot", "Creating External Snapshot of guest's disk";
+        script_run("rm -f /var/lib/libvirt/images/$guest.{disk-only,memspec,diskspec}");    # ensure the files do not exist already
```

Comment:
Quick question,
Why do `rm -f /var/lib/libvirt/images/$guest.{disk-only,memspec,diskspec}` before creating the External Snapshot of guest's disk

it did trigger virsh_external_snapshot.pm - create external snapshot of guest disk post on this worker, right ?

Reply:
refer to comments - `# ensure the files do not exist already`
would be
`if (script_run("file /var/lib/libvirt/images/$guest.{disk-only,memspec,diskspec}") == 0) {script_run("rm -f /var/lib/libvirt/images/$guest.{disk-only,memspec,diskspec}");}`

Reply:
Thanks for the comment.

We are doing this to ensure, that no external snapshots exist, otherwise this test fails. This is required if we need to restart the test.

`rm -f` does not fail if the files do not exists, so I think we don't need to check for the files beforehand as @guoxuguang suggest. You are right though, the files will not exists, but with `rm -f` the command succeeds and does not complain.

IMHO it should be fine this way or am I missing something?

Reply:
sure, make sense here.

imho, we would be better to detect the files here firstly - make sure the current testing environment clear
if so - trigger rm -f command
if not - follow the next external snapshots command